Cannot run any OpenAI client chat completions with model from latest docker image

### System Info

Lorax: v0.8.1
Latest Docker Build: 
```
docker pull ghcr.io/predibase/lorax:sha256-6207ab8714fe9f56adf05a7a9077f41d9d0c5b706a7f718c35493aef0a514dec
```

Using the default latest docker build throws an error that it cannot find 'adapter_config.json' when attempting to access the API via the OpenAI python client, even though no adapters or loras are suggested or provided via the API. Access through 127.0.0.1:8080/generate works as expected. 

It may be related to this issue: https://github.com/predibase/lorax/issues/315. 

The docker run command is copied verbatim from the readme, and the python client code is copied directly from the readme and only modified the model name to match the docker run command. Sample code is provided. 

Model: 
```
{
  "model_id": "mistralai/Mistral-7B-Instruct-v0.1",
  "model_sha": "73068f3702d050a2fd5aa2ca1e612e5036429398",
  "model_dtype": "torch.float16",
  "model_device_type": "cuda",
  "model_pipeline_tag": "text-generation",
  "max_concurrent_requests": 128,
  "max_best_of": 2,
  "max_stop_sequences": 4,
  "max_input_length": 1024,
  "max_total_tokens": 2048,
  "waiting_served_ratio": 1.2,
  "max_batch_total_tokens": 47088,
  "max_waiting_tokens": 20,
  "validation_workers": 2,
  "version": "0.1.0",
  "sha": null,
  "docker_label": null
}
```


### Information

- [X] Docker
- [ ] The CLI directly

### Tasks

- [X] An officially supported command
- [ ] My own modifications

### Reproduction

1. Run the docker command verbatim from the Readme: 
```bash
model=mistralai/Mistral-7B-Instruct-v0.1
volume=$PWD/data

docker run --gpus all --shm-size 1g -p 8080:80 -v $volume:/data \
    ghcr.io/predibase/lorax:latest --model-id $model
```
2. Call the docker from python script copied verbatim from the readme with the model docker run command in 1 above: "mistralai/Mistral-7B-Instruct-v0.1"

```python
from openai import OpenAI

client = OpenAI(
    api_key="EMPTY",
    base_url="http://127.0.0.1:8080/v1",
)

resp = client.chat.completions.create(
    model="mistralai/Mistral-7B-Instruct-v0.1",
    messages=[
        {
            "role": "system",
            "content": "You are a friendly chatbot who always responds in the style of a pirate",
        },
        {"role": "user", "content": "How many helicopters can a human eat in one sitting?"},
    ],
    max_tokens=100,
)
print("Response:", resp.choices[0].message.content)
```

3. Observe error: 
```
2024-03-14T12:53:36.386646Z ERROR lorax_launcher: server.py:158 Error when downloading adapter
```

Stacktrace logged from docker:
```
Entry Not Found for url: https://huggingface.co/mistralai/Mistral-7B-Instruct-v0.1/resolve/main/adapter_config.json.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/bin/lorax-server", line 8, in <module>
    sys.exit(app())
  File "/opt/conda/lib/python3.10/site-packages/typer/main.py", line 311, in __call__
    return get_command(self)(*args, **kwargs)
  File "/opt/conda/lib/python3.10/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/opt/conda/lib/python3.10/site-packages/typer/core.py", line 778, in main
    return _main(
  File "/opt/conda/lib/python3.10/site-packages/typer/core.py", line 216, in _main
    rv = self.invoke(ctx)
  File "/opt/conda/lib/python3.10/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/conda/lib/python3.10/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/conda/lib/python3.10/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/opt/conda/lib/python3.10/site-packages/typer/main.py", line 683, in wrapper
    return callback(**use_params)  # type: ignore
  File "/opt/conda/lib/python3.10/site-packages/lorax_server/cli.py", line 89, in serve
    server.serve(
  File "/opt/conda/lib/python3.10/site-packages/lorax_server/server.py", line 299, in serve
    asyncio.run(
  File "/opt/conda/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/opt/conda/lib/python3.10/asyncio/base_events.py", line 636, in run_until_complete
    self.run_forever()
  File "/opt/conda/lib/python3.10/asyncio/base_events.py", line 603, in run_forever
    self._run_once()
  File "/opt/conda/lib/python3.10/asyncio/base_events.py", line 1909, in _run_once
    handle._run()
  File "/opt/conda/lib/python3.10/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "/opt/conda/lib/python3.10/site-packages/grpc_interceptor/server.py", line 165, in invoke_intercept_method
    return await self.intercept(
> File "/opt/conda/lib/python3.10/site-packages/lorax_server/interceptor.py", line 38, in intercept
    return await response
  File "/opt/conda/lib/python3.10/site-packages/opentelemetry/instrumentation/grpc/_aio_server.py", line 82, in _unary_interceptor
    raise error
  File "/opt/conda/lib/python3.10/site-packages/opentelemetry/instrumentation/grpc/_aio_server.py", line 73, in _unary_interceptor
    return await behavior(request_or_iterator, context)
  File "/opt/conda/lib/python3.10/site-packages/lorax_server/server.py", line 154, in DownloadAdapter
    PeftConfig.from_pretrained(config_path, token=api_token)
  File "/opt/conda/lib/python3.10/site-packages/peft/utils/config.py", line 121, in from_pretrained
    raise ValueError(f"Can't find '{CONFIG_NAME}' at '{pretrained_model_name_or_path}'")
ValueError: Can't find 'adapter_config.json' at 'mistralai/Mistral-7B-Instruct-v0.1'
```

### Expected behavior

The python script fails to execute due to a server side error that complains about the 'adapter_config.json' not being able to load. Note the python client code does NOT specify any adapter at all. 

It should return the generated text in the expected OpenAI API-compatible format so that 
```python
print("Response:", resp.choices[0].message.content)
```
prints the generated text.

--- a/router/src/server.rs
+++ b/router/src/server.rs
@@ -118,7 +118,7 @@ async fn completions_v1(
     let mut req = req.0;
     if req.model == MODEL_ID.get().unwrap().as_str() {
         // Allow user to specify the base model, but treat it as an empty adapter_id
-        tracing::debug!("Replacing base model {0} with empty adapter_id", req.model);
+        tracing::info!("Replacing base model {0} with empty adapter_id", req.model);
         req.model = "".to_string();
     }
     let mut gen_req = CompatGenerateRequest::from(req);
@@ -184,7 +184,7 @@ async fn chat_completions_v1(
     let mut req = req.0;
     if req.model == MODEL_ID.get().unwrap().as_str() {
         // Allow user to specify the base model, but treat it as an empty adapter_id
-        tracing::debug!("Replacing base model {0} with empty adapter_id", req.model);
+        tracing::info!("Replacing base model {0} with empty adapter_id", req.model);
         req.model = "".to_string();
     }
     let mut gen_req = CompatGenerateRequest::from(req);

--- a/router/src/server.rs
+++ b/router/src/server.rs
@@ -115,7 +115,12 @@ async fn completions_v1(
     req_headers: HeaderMap,
     req: Json<CompletionRequest>,
 ) -> Result<Response, (StatusCode, Json<ErrorResponse>)> {
-    let req = req.0;
+    let mut req = req.0;
+    if req.model == MODEL_ID.get().unwrap().as_str() {
+        // Allow user to specify the base model, but treat it as an empty adapter_id
+        tracing::debug!("Replacing base model {0} with empty adapter_id", req.model);
+        req.model = "".to_string();
+    }
     let mut gen_req = CompatGenerateRequest::from(req);
 
     // default return_full_text given the pipeline_tag
@@ -176,7 +181,12 @@ async fn chat_completions_v1(
     req_headers: HeaderMap,
     req: Json<ChatCompletionRequest>,
 ) -> Result<Response, (StatusCode, Json<ErrorResponse>)> {
-    let req = req.0;
+    let mut req = req.0;
+    if req.model == MODEL_ID.get().unwrap().as_str() {
+        // Allow user to specify the base model, but treat it as an empty adapter_id
+        tracing::debug!("Replacing base model {0} with empty adapter_id", req.model);
+        req.model = "".to_string();
+    }
     let mut gen_req = CompatGenerateRequest::from(req);
 
     // default return_full_text given the pipeline_tag

--- a/server/lorax_server/server.py
+++ b/server/lorax_server/server.py
@@ -144,37 +144,24 @@ class LoraxService(generate_pb2_grpc.LoraxServiceServicer):
             if adapter_source == PBASE:
                 adapter_id = map_pbase_model_id_to_s3(adapter_id, api_token)
                 adapter_source = S3
-            try:
-                if adapter_source == HUB:
-                    # Quick auth check on the repo against the token
-                    HfApi(token=api_token).model_info(adapter_id, revision=None)
-                    
-                    # fail fast if ID is not an adapter (i.e. it is a full model)
-                    # TODO(geoffrey): do this for S3– can't do it this way because the
-                    # files are not yet downloaded locally at this point.
-                    config_path = get_config_path(adapter_id, adapter_source)
-                    PeftConfig.from_pretrained(config_path, token=api_token)
+            
+            if adapter_source == HUB:
+                # Quick auth check on the repo against the token
+                HfApi(token=api_token).model_info(adapter_id, revision=None)
+                
+                # fail fast if ID is not an adapter (i.e. it is a full model)
+                # TODO(geoffrey): do this for S3– can't do it this way because the
+                # files are not yet downloaded locally at this point.
+                config_path = get_config_path(adapter_id, adapter_source)
+                PeftConfig.from_pretrained(config_path, token=api_token)
 
-                _download_weights(
-                    adapter_id, source=adapter_source, api_token=api_token
-                )
+            _download_weights(
+                adapter_id, source=adapter_source, api_token=api_token
+            )
 
-                # Calculate size of adapter to be loaded
-                source = get_model_source(adapter_source, adapter_id, extension=".safetensors", api_token=api_token)
-                adapter_bytes += source.get_weight_bytes()
-            except Exception:
-                logger.exception("Error when downloading adapter")
-
-                if adapter_source != LOCAL:
-                    # delete safetensors files if there is an issue downloading or converting 
-                    # the weights to prevent cache hits by subsequent calls
-                    try:
-                        local_path = get_local_dir(adapter_id, adapter_source)
-                        shutil.rmtree(local_path)
-                    except Exception as e:
-                        logger.warning(f"Error cleaning up safetensors files after "
-                                       f"download error: {e}\nIgnoring.")
-                raise
+            # Calculate size of adapter to be loaded
+            source = get_model_source(adapter_source, adapter_id, extension=".safetensors", api_token=api_token)
+            adapter_bytes += source.get_weight_bytes()
         
         adapter_memory_size = self.model.adapter_memory_size()
         if adapter_memory_size > 0:

